layer2/ObjectMap.cpp

```diff
@@ -5956,3 +5956,58 @@ ObjectMap *ObjectMapLoadChemPyMap(PyMOLGlobals * G, ObjectMap * I, PyObject * Ma
   return (I);
 #endif
 }
+
+void ObjectMapDump(ObjectMap* om, const char* fname, int state, int quiet)
+{
+  auto file = fopen(fname, "wb");
+  if (!file) {
+    ErrMessage(om->G, "ObjectMapDump", "can't open file for writing");
+  }
+  
+  ObjectMapState* oms = om->State + 0;
+  
+  int zd = oms->Field->dimensions[2];
+  int yd = oms->Field->dimensions[1];
+  int xd = oms->Field->dimensions[0];
+  
+  float xs = (oms->ExtentMax[0] - oms->ExtentMin[0]) / (oms->Field->dimensions[0]);
+  float ys = (oms->ExtentMax[1] - oms->ExtentMin[1]) / (oms->Field->dimensions[1]);
+  float zs = (oms->ExtentMax[2] - oms->ExtentMin[2]) / (oms->Field->dimensions[2]);
+  
+  int offset = 0;
+  for (int xi = 0; xi < oms->Field->dimensions[0]; xi++) {
+    for (int yi = 0; yi < oms->Field->dimensions[1]; yi++) {
+      for (int zi = 0; zi < oms->Field->dimensions[2]; zi++) {
+        float x = oms->ExtentMin[0] + xs * xi;
+        float y = oms->ExtentMin[1] + ys * yi;
+        float z = oms->ExtentMin[2] + zs * zi;
+        
+        CField* field = oms->Field->data;
+        
+        if(field->type == cFieldFloat) {
```

Comment:
Not sure if this if/else block is entirely reliable since the size of types can be different across platforms, and we could incidentally have user-defined types that have the same size as built-in types. As far as I understand, we only have Field data of `int`, `float`, or `Tetsurf::PointType`/`IsoSurf::PointType`. For now, might be best to just query `cFieldFloat`, `cFieldInt`, and probably just ignore the rest. `Field` of `ObjectMapState` seem to only have underlying `float` anyway...

Reply:
So maybe stick only with `float`? If so then also get rid of `field->type` seems alright.

Would be reliable because importing and exporting already rely on float (eg `ObjectMapXPLORStrToMap`). Also because sessions rely on Python objects that don't have such kind of problems (eg `ObjectMapStateAsPyList`).

The options would be:

```
        switch (field->type) {
          case cFieldFloat: value = (float)((float*)field)[offset]; break;
          case cFieldInt: value = (double)((int*)field)[offset]; break;
          default:
            ErrMessage(om->G, "ObjectMapDump", "unknown field type");
        }
```

or 
 ```
        float value = (float)((float*)field)[offset];
```

Reply:
Maybe I talked nonsense.